src/map/script.c

```diff
@@ -5685,24 +5693,64 @@ BUILDIN(goto)
 	return true;
 }
 
-/*==========================================
- * user-defined function call
- *------------------------------------------*/
+/// User-defined function call and subroutine call (Updated by [Cretino]) (Thanks to [FlavioJS])
+/// 'callfunc' support labels in the current npc "::LabelName"
+/// 'callfunc' support labels in other npcs "NpcName::LabelName"
 BUILDIN(callfunc)
 {
-	int i, j;
-	struct script_retinfo* ri;
-	struct script_code* scr;
-	const char* str = script_getstr(st,2);
+	int i, j, k, pos = -1;
+	struct script_retinfo *ri;
+	struct script_code *scr = NULL;
 	struct reg_db *ref = NULL;
+	struct npc_data *nd = NULL;
+	struct script_data *data = script_getdata(st, 2);
 
-	scr = (struct script_code*)strdb_get(script->userfunc_db, str);
-	if( !scr )
-	{
-		ShowError("script:callfunc: function not found! [%s]\n", str);
-		st->state = END;
-		return false;
-	}
+	if( !data_islabel(data) && !data_isfunclabel(data) ) {
+		const char *str = script_getstr(st, 2);
+		scr = (struct script_code*)strdb_get(script->userfunc_db, str);
+
+		if (!scr) {
+			if (stristr(str, "::")) {
+				char labelname[NAME_LENGTH], npcname[NAME_LENGTH];
+
+				memset(labelname, '\0', sizeof(labelname));
+				memset(npcname, '\0', sizeof(npcname));
```

Comment:
No need to memset at all. You could simply do:

char labelname[NAME_LENGTH] = "", npcname[NAME_LENGTH] = "";